autoblocks/_impl/tracerv2/run.py

```diff
@@ -0,0 +1,263 @@
+            try:
+                evaluation = await evaluator.evaluateOnline(event=event)
+            except Exception:
+                log.error("Do something here")
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await loop.run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate,
+                    event,
+                )
+            except Exception:
+                log.error("#TODO do something here")
+
+        if evaluation is None:
+            return
+
+        return evaluation
+
+    async def _send_event_unsafe(
+        self,
+        # Require all arguments to be specified via key=value
+        *,
+        message: str,
+        trace_id: Optional[str] = None,
+        span_id: Optional[str] = None,
+        parent_span_id: Optional[str] = None,
+        timestamp: Optional[str] = None,
+        properties: Optional[Dict] = None,
+        evaluators: List[BaseEvaluator] = [],
+        max_evaluator_concurrency: int = 5,
+    ) -> SendEventResponse:
+        merged_properties = dict(self._properties)
+        merged_properties.update(properties or {})
+        if span_id:
+            merged_properties["span_id"] = span_id
+        if parent_span_id:
+            merged_properties["parent_span_id"] = parent_span_id
+
+        trace_id = trace_id or self._trace_id
+        timestamp = timestamp or datetime.now(datetime.timezone.utc).isoformat()
+
+        try:
+            replay_headers = make_replay_headers()
+        except Exception as err:
+            log.error(f"Failed to generate replay headers: {err}", exc_info=True)
+            replay_headers = None
+
+        event_dict = {
+            "message": message,
+            "traceId": trace_id,
+            "timestamp": timestamp,
+            "properties": merged_properties,
+        }
+
+        evaluations = []
+        try:
+            evaluations = await gather_with_max_concurrency(
+                max_evaluator_concurrency,
+                [
+                    self._evaluate_event(
+                        event=event_dict,
+                        evaluator=evaluator,
+                    )
+                    for evaluator in evaluators
+                ],
+            )
+        except Exception:
+            log.error("TODO: DO something here")
+
+        print(evaluations)
+        req = self._client.post(
+            url=INGESTION_ENDPOINT,
+            json={
+                "message": message,
+                "traceId": trace_id,
+                "timestamp": timestamp,
+                "properties": merged_properties,
+            },
+            headers=replay_headers,
+            timeout=self._timeout_seconds,
+        )
+        req.raise_for_status()
+        resp = req.json()
+        return SendEventResponse(trace_id=resp.get("traceId"))
+
+    async def async_send_event(
```

Comment:
i don't think we need this level right? can just rename `_send_event_unsafe` to `_async_send_event` and call it from the main `send_event`